### ISSUE TITLE
Improvement: Need a mechanism to create custom nodes that DOESN'T save them to the library

### ISSUE THREAD

**Issue by AndrewHeumann** [DynamoDS/Dynamo#6633]
Because of the nature of list management in Dynamo, it is frequently functionally necessary to take some group of nodes and turn them into a custom node for the purposes of list.mapping or otherwise. However, creating a custom node ALWAYS means saving that node to the user library. As such I have an overflowing list of one-off junk in a "throwaway" category in my node library - stuff I could never conceivably reuse because it's too specific, but dynamo insists on saving it anyway. Custom Node creation and saving reusable custom nodes should be _separate_ functions (See how Grasshopper distinguishes UserObjects from Clusters - a cluster can be a user object but not all clusters are user objects.) 
![image](https://cloud.githubusercontent.com/assets/2157902/15300657/be3dc44c-1b77-11e6-9faf-13cb6acc8166.png)

**Comment by dimven** [comment]
See https://github.com/DynamoDS/Dynamo/issues/6634

Consider saving that one-off CN in the same folder as the graph file. (instead of the user's library) That way it will show in the library only when you open that specific file.

If a mechanism is created to prevent a CN from appearing in the library, you'd need to invent an alternative method for placing new instances of that CN on canvas.

**Comment by AndrewHeumann** [comment]
but @dimven I can't even create a CN WITHOUT it asking where to add it in the library. it shows up in my library as soon as it is created. Even if I do not save the .dyf file it spawns, or save it as something else, it is already created and added to my library.

and I don't think an "alternative method for placing new instances" is necessary - the whole point is that these nodes are not really reusable.

**Comment by dimven** [comment]
I see what you mean. So basically you're looking for a truly one-off node that lives only in the canvas (isn't stored anywhere else and is tied to the *dyn file), does not appear in the library and does not need to be placed multiple times, a.k.a. a model-in-place node ?

**Comment by AndrewHeumann** [comment]
Yeah exactly. For example, a large but specific code block function that i want to "list.map" - I cannot do so unless I "custom node" it.

**Comment by mjkkirschner** [comment]
You can map it using some funky syntax to build a function object from a function pointer - will post example later but I see what you want, something like an anonymous function. I'd like it as well.😜

**Comment by andydandy74** [comment]
@AndrewHeumann - have you tried working with ```Function.Compose``` and ```Function.Apply```? Maybe worth a try.

**Comment by AndrewHeumann** [comment]
Thanks Andreas, didn't know about those - but doesn't change the need for my initial request. In grasshopper I use clusters (the equivalent of custom nodes) all the time (even for purely organizational purposes) without the desire to save them for future re-use in other graphs.

**Comment by Curty102** [comment]
+1
